File: app/src/main/java/me/wcy/music/adapter/PlayPagerAdapter.java
package me.wcy.music.adapter;

import android.support.v4.view.PagerAdapter;
import android.view.View;
import android.view.ViewGroup;

import java.util.List;

/**
 * 正在播放ViewPager适配器，包含歌词和封面
 * Created by wcy on 2015/11/30.
 */
public class PlayPagerAdapter extends PagerAdapter {
    private List<View> mViews;

    public PlayPagerAdapter(List<View> views) {
        mViews = views;
    }

    @Override
    public int getCount() {
        return mViews.size();
    }

    @Override
    public boolean isViewFromObject(View view, Object object) {
        return view == object;
    }

    @Override
    public Object instantiateItem(ViewGroup container, int position) {
        container.addView(mViews.get(position));
        return mViews.get(position);
    }

    @Override
    public void destroyItem(ViewGroup container, int position, Object object) {
        container.removeView(mViews.get(position));
    }
}

File: app/src/main/java/me/wcy/music/adapter/PlayPagerAdapter.kt
package me.wcy.music.adapter

import android.view.View
import android.view.ViewGroup
import androidx.viewpager.widget.PagerAdapter

/**
 * 正在播放ViewPager适配器，包含歌词和封面
 * Created by wcy on 2015/11/30.
 */
class PlayPagerAdapter(private val mViews: List<View>) : PagerAdapter() {
    override fun getCount(): Int {
        return mViews.size
    }

    override fun isViewFromObject(view: View, `object`: Any): Boolean {
        return view === `object`
    }

    override fun instantiateItem(container: ViewGroup, position: Int): Any {
        container.addView(mViews[position])
        return mViews[position]
    }

    override fun destroyItem(container: ViewGroup, position: Int, `object`: Any) {
        container.removeView(mViews[position])
    }
}